swupd/manifest.go

```diff
@@ -853,6 +856,9 @@ func writeIndexManifest(c *config, ui *UpdateInfo, bundles []*Manifest) (*Manife
 	oldM.sortFilesName()
 	// linkPeersAndChange will update file versions correctly
 	_, _, _ = idxMan.linkPeersAndChange(oldM, ui.minVersion)
+
+	idxMan.Header.FileCount = uint32(len(idxMan.Files))
```

Comment:
Why do you need to set this again here? Does it get overwritten after line 819?

Reply:
I intended to set the file count on 860 instead of 819, so I will remove the reference on 819.

When a file is deleted `linkPeersAndChange` will add a deleted file entry, but it doesn't update `idxMan.Header.FileCount`, so this value would be incorrect when deleting bundles (it leads to a failure when deleting the index manifest). Also, in the typical, bundle workflow, `idxMan.Header.FileCount` is set after `linkPeersAndChange`